### PR TITLE
feat(nextflow): add secrets subcommand

### PR DESCRIPTION
**Pull request by mribeirodantas**
- feat(nextflow): Add secrets subcommand
- feat(nextflow): Add generator for secret names

**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
It adds autocomplete to the secrets subcommand in the nextflow CLI and a generator
for autocompleting the argument of the `nextflow secrets get` subcommand.

### CODE REVIEW

File: src/nextflow.ts [hunk 1]
```diff
@@ -25,6 +25,21 @@ const projectname: Fig.Generator = {
   },
 };
 
+const secretname: Fig.Generator = {
+  script: `grep -o '"name": *"[^"]*"' $HOME/.nextflow/secrets/store.json | grep -o '"[^"]*"$' | tr -d \\"`,
+  postProcess: (output) => {
```
Comment (fedeci) [inline src/nextflow.ts:30]: Could this be implemented using "nextflow secrets list" as the script?

File: src/nextflow.ts [hunk 2]
```diff
@@ -918,6 +933,40 @@ const completionSpec: Fig.Spec = {
...
+          description:
+            "Allows creating a new secret or overriding an existing one e.g. nextflow secrets set FOO 'Hello world'",
+          args: {
+            name: "secret name and value",
+          },
+        },
+        {
+          name: "delete",
+          description:
+            "Allows deleting an existing secret e.g. nextflow secrets delete FOO",
```
Comment (fedeci) [inline src/nextflow.ts:966]:
```suggestion
            "Allows deleting an existing secret e.g. nextflow secrets delete FOO",
          isDangerous: true,
          args: {
             name: "secret name",
             generators: secretname,
           },
```

File: src/nextflow.ts [hunk 3]
```diff
@@ -918,6 +933,40 @@ const completionSpec: Fig.Spec = {
...
+            name: "secret name",
+            generators: secretname,
+          },
+        },
+        {
+          name: "set",
+          description:
+            "Allows creating a new secret or overriding an existing one e.g. nextflow secrets set FOO 'Hello world'",
+          args: {
+            name: "secret name and value",
```
Comment (fedeci) [inline src/nextflow.ts:960]:
```suggestion
            name: "secret name and value",
            generators: secretname,
            suggestCurrentToken: true,
```

File: src/nextflow.ts [hunk 4]
```diff
@@ -918,6 +933,40 @@ const completionSpec: Fig.Spec = {
...
+            "Allows retrieving a secret value e.g. nextflow secrets get FOO",
+          args: {
+            name: "secret name",
+            generators: secretname,
+          },
+        },
+        {
+          name: "set",
+          description:
+            "Allows creating a new secret or overriding an existing one e.g. nextflow secrets set FOO 'Hello world'",
```
Comment (fedeci) [inline src/nextflow.ts:958]:
```suggestion
            "Allows creating a new secret or overriding an existing one e.g. nextflow secrets set FOO 'Hello world'",
          isDangerous: true,
```

**Review by fedeci — CHANGES_REQUESTED** [review]

File: src/nextflow.ts (hunk 1 above)
Comment (mribeirodantas) [reply src/nextflow.ts:30]: Unfortunately, I'm unable to run nextflow commands in the script field of generators on Fig. That's what I get:

```
The operation couldn’t be completed. Unable to locate a Java Runtime.
Please visit http://www.java.com for information on installing Java.

[31m/Users/mribeirodantas/.local/bin/nextflow: line 287: /bin/java: No such file or directory(B[m
[33mNOTE: Nextflow is trying to use the Java VM defined by the following environment variables:
 JAVA_CMD: /bin/java
 NXF_OPTS: 
(B[m
```

That's why I'm always using ShellScript to scrap the info for the generators.

**Comment by mribeirodantas** [comment]
Is there a bot in this repository to automatically fix linting issues? Something like commenting `@bot-this fix linting` and it'd commit the fix.

**Comment by SeparateRecords** [comment]
> Is there a bot in this repository to automatically fix linting issues? Something like commenting @bot-this fix linting and it'd commit the fix.

This isn't implemented but we have a command to fix it:
```bash
yarn lint:fix
# ... or `npm run lint:fix`, etc.
```

**Comment by mribeirodantas** [comment]
Yeah, I knew this one. Thanks anyway! :)

**Comment by fedeci** [comment]
> Is there a bot in this repository to automatically fix linting issues? Something like commenting `@bot-this fix linting` and it'd commit the fix.

Sounds like a good addition! Do you know any OSS project using a bot like that?

**Comment by mribeirodantas** [comment]
> Sounds like a good addition! Do you know any OSS project using a bot like that?

Sure. We have `@nf-core-bot fix linting` in https://github.com/nf-core/nf-co.re.
Check here, for example: https://github.com/nf-core/nf-co.re/pull/1404#issuecomment-1285162283

**Comment by mribeirodantas** [comment]
Oh, my... I did something wrong on my side 🤦

File: src/nextflow.ts (hunk 1 above)
Comment (SeparateRecords) [reply src/nextflow.ts]: Maybe for the script you could try `"sh -c 'JAVA_CMD=$(which java) ...'"` where ... is the logic? Not sure if that would work.

**Pull request closed without merging**